arangod/Graph/Enumerators/TwoSidedEnumerator.h

```diff
@@ -192,6 +263,22 @@ class TwoSidedEnumerator {
    */
   auto stealStats() -> aql::TraversalStats;
 
+  auto setForbiddenVertices(std::unique_ptr<VertexSet> forbidden)
+      -> void requires HasForbidden<PathValidatorType> {
+    auto copy = std::make_unique<VertexSet>(*forbidden);
+    _left.setForbiddenVertices(std::move(copy));
+    _right.setForbiddenVertices(std::move(forbidden));
+  };
+
```

Comment:
Why not make it a shared_ptr? How does this work, doesn't VertexSet gets destroyed two times then?

Reply:
I will make it a `shared_ptr`, you are right. What we have works but is less efficient, since it copies the whole set.